### PR TITLE
Add torch.tensor replacement and int_tensor prim

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88221

### CODE REVIEW

File: torch/_prims/__init__.py [hunk 1]
```diff
@@ -2541,6 +2542,18 @@ def _scalar_tensor_aten(
...
 )
 
+_int_tensor_doc = """
+    Wraps a SymInt or a list of SymInts into a Tensor with the specified dtype and device.
+"""
+
+int_tensor = _make_prim(
+    # TODO: handle scalar values
+    schema="int_tensor(SymInt[] data, *, ScalarType? dtype=None, Device? device=None, bool pin_memory=False, bool requires_grad=False) -> Tensor",
+    meta=torch.tensor,
```
Comment (ezyang) [inline torch/_prims/__init__.py:2552]: This meta is too permissive; you need to only accept lists of ints as the input, arbitrary torch.tensor inputs are not OK.

File: torch/_refs/__init__.py [hunk 2]
```diff
@@ -4482,6 +4483,32 @@ def scalar_tensor(
...
+    len_ = None
+    for item in a:
+        if isinstance(item, Iterable):
+            return False
+        if isinstance(item, torch.SymInt):
+            return True
+    return False
+
+
+def torch_tensor_r(
```
Comment (ezyang) [inline torch/_refs/__init__.py:4497]: why isn't this just called `tensor`

File: torch/_refs/__init__.py [hunk 3]
```diff
@@ -4482,6 +4483,32 @@ def scalar_tensor(
...
+    *,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+    requires_grad: bool = False,
+) -> TensorLikeType:
+    utils.check_pin_memory(pin_memory)
+    # dtype = dtype if dtype is not None else utils.type_to_dtype(type(a))
+    device = device if device is not None else torch.device("cpu")
+    if isinstance(a, torch.SymInt) or check_for_1d_list_with_symints(a):
```
Comment (ezyang) [inline torch/_refs/__init__.py:4508]: You can support ND lists; just flatten the list before passing it to `int_tensor`

File: torch/_prims/__init__.py (hunk 1 above)
Comment (anjali411) [reply torch/_prims/__init__.py:2552]: that's already enforced by the schema

File: torch/_refs/__init__.py (hunk 3 above)
Comment (anjali411) [reply torch/_refs/__init__.py:4508]: I can but there are some checks we do to ensure data has a shape that's ok (consistent length of data with each dim). Skipped it for now but can add a similar function in python

File: torch/_prims/__init__.py [hunk 4]
```diff
@@ -2541,6 +2542,18 @@ def _scalar_tensor_aten(
     doc=_scalar_tensor_doc,
 )
 
+_int_tensor_doc = """
+    Wraps a SymInt or a list of SymInts into a Tensor with the specified dtype and device.
+"""
+
+int_tensor = _make_prim(
+    # TODO: handle scalar values
+    schema="int_tensor(SymInt[] data, *, ScalarType? dtype=None, Device? device=None, bool pin_memory=False, bool requires_grad=False) -> Tensor",
```
Comment (ezyang) [inline torch/_prims/__init__.py:2551]: It will be best if the prim takes shape in; otherwise you will have to expose unsafe_view to review the tensor to the nd shape after

File: torch/_refs/__init__.py [hunk 5]
```diff
@@ -4482,6 +4483,32 @@ def scalar_tensor(
...
     return prims.scalar_tensor(a, dtype=dtype, device=device)
 
+def check_for_1d_list_with_symints(a):
+    assert isinstance(a, Iterable)
+    len_ = None
+    for item in a:
+        if isinstance(item, Iterable):
+            return False
+        if isinstance(item, torch.SymInt):
+            return True
```
Comment (ezyang) [inline torch/_refs/__init__.py:4493]: What happens if I pass torch.tensor([symint. 2.0])

**Comment by ezyang** [comment]
Here's what needs to be done next:

- Write a simple test case to `test_proxy_tensor.py` that exercises the error case. Something like `return torch.tensor([tensor.size(0)])`
- Patch torch.tensor so that it runs your ref, instead of the regular implementation. This can be done with just monkeypatching, or with a TorchFunctionMode.
- Show that the test case now correctly traces. Once the test case is working, show that the model you were working on is also working

**Comment by ezyang** [comment]
un-staling this, we still need it, but someone else will have to finish it up
